src/data-table/DataTable.tsx

```diff
@@ -145,11 +147,16 @@ export function DataTable<T extends ReferenceObject = TableObject>(props: DataTa
     );
 
     useEffect(() => updatePagination(pagination => ({ ...pagination, page: 1 })), [resetKey]);
+
     useEffect(() => {
         const newVisibleColumns = columns.filter(({ hidden }) => !hidden).map(({ name }) => name);
         updateVisibleColumns(visibleColumns => _.uniq([...visibleColumns, ...newVisibleColumns]));
     }, [columns]);
 
+    useEffect(() => {
+        if (onReorderColumns) onReorderColumns(visibleColumns);
+    }, [visibleColumns]);
```

Comment:
`onReorderColumns` should be a dependency?

Reply:
Yes